### ISSUE TITLE
Fix inconsistent techniques of binding JavaScript functions

### ISSUE THREAD

**Issue by Ishirak** [versionpress/versionpress#1090]
This issue is part of a frontend code review, see #1083.

Three binding techniques are used in the front end codebase.
- In the constructor.
- In the render method.
- Not binding at all – functions are in component's private attributes.

Binding in the constructor is a valid option and is used by many developers. 
Binding in the render method is also used by many developers, but is not a performant option – a bind call in the render method creates a brand new function on every single render. ([MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind))

My preffered solution is to use [arrow functions] (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions). They don't require boilerplate in the constructor, stylistically aren't much different and automatically bind themselves.

For example, constructor binding looks like this:
```JSX
    // ...
    constructor() {
        // ...
        this.onClearSelection = this.onClearSelection.bind(this);
        // ...
    }

    onClearSelection(e: React.MouseEvent) {
        // ...
    }
    // ...
```
On the other hand, arrow functions look like this:
```JSX
    // ...
    onClearSelection = (e: React.MouseEvent) => {
        // ...
    }
    // ...
```

**Comment by vasek17** [comment]
I'm not used to [class instance fields](https://github.com/jeffmo/es-class-fields-and-static-properties) but it looks it will not be changed in the future so I think it is OK to use it.

The same goes for initializing the state - instead of
```javascript
class Foo extends React.Component {
    // ...
    constructor(props) {
        super(props);
        this.state = { ... }
    }
    // ...
}
```

we could write

```javascript
class Foo extends React.Component {
    // ...
    state = { ... }
    // ...
}
```

**Comment by borekb** [comment]
:+1:

**Comment by Ishirak** [comment]
:+1: I'm currently using this form of state initialization in my projects, it's much cleaner.

**Comment by borekb** [comment]
:ship: Shipped in [4.0-alpha1](https://github.com/versionpress/versionpress/releases/tag/4.0-alpha1).
